.github/workflows/benchmark.yml

```diff
@@ -69,31 +69,43 @@ jobs:
           name: body
           path: ci-benchmark/body.md
 
-  Cache_Benchmarks:
-    name: Cache Micro Benchmarks result
+  Track_and_Cache_Benchmarks:
+    name: Track and Cache Micro Benchmarks
     if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
     permissions:
       pull-requests: write
       contents: write
     runs-on: benchmarking-runner
+    env:
+      BENCHER_PROJECT: tailcall
+      BENCHER_ADAPTER: rust_criterion
+      BENCHER_TESTBED: benchmarking-runner
+      BASE_BENCHMARK_RESULTS: benches/base.txt
     steps:
       - name: Check out code
         uses: actions/checkout@v4
 
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
 
-      - name: Run Benchmarks
-        run: |
-          cargo install cargo-criterion rust-script
-          cargo criterion --message-format=json > benches/main_benchmarks.json
-          ./scripts/json_to_md.rs benches/main_benchmarks.json > benches/main_benchmarks.md
-          cat benches/main_benchmarks.md
+      - name: Install Bencher CLI
+        uses: bencherdev/bencher@main
 
-      - name: Cache Criterion Benchmarks Json
+      - name: Run and track Benchmarks
+        run: |
+          cargo bench > benches/base.txt
+          bencher run \
+          --project "$BENCHER_PROJECT" \
+          --branch main \
```

Comment:
Use the new `--branch-reset` option: https://bencher.dev/docs/explanation/branch-selection/#--branch-reset

See: https://bencher.dev/docs/how-to/track-benchmarks/#relative-continuous-benchmarking

Reply:
` Run and track Benchmarks ` job is to track the benchmarks so we have a  performance history of the benchmarks.
Using --branch-reset would recreate the branch and discontinue the tracking.
I think you meant using  `--branch-reset` in the [criterion_compare](https://github.com/tailcallhq/tailcall/pull/1679/files#diff-aacbd0338bcf598920c120b69adaba2672acc7ea314cfca0f792bebe5fe2acd0L99) job